src/api/bdaddr.rs

```diff
@@ -203,6 +193,163 @@ impl BDAddr {
     }
 }
 
+/// Different de-/serialization formats for [`BDAddr`].
+#[cfg(feature = "serde")]
+pub mod serde {
+    use std::fmt::{self, Write as _};
+
+    use serde::{
+        de::{Deserialize, Deserializer, Error as DeError, Visitor},
+        ser::{Serialize, Serializer},
+    };
+    use serde_cr as serde;
+
+    use super::*;
+
+    impl Serialize for BDAddr {
+        fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+        where
+            S: Serializer,
+        {
+            colon_delim::serialize(self, serializer)
+        }
+    }
+
+    impl<'de> Deserialize<'de> for BDAddr {
+        fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+        where
+            D: Deserializer<'de>,
+        {
+            colon_delim::deserialize(deserializer)
+        }
+    }
+
+    /// De-/Serialization of [`BDAddr`] as string of hex-digits with colons as delimiters.
+    ///
+    /// This is the standard used to de-/seriallize [`BDAddr`].
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// //use btleplug::api::BDAddr;
+    /// //use serde::{Serialize, Deserialize};
+    /// //^^^ Error can't find crate `serde`???
+    /// //# use serde_cr as serde;
+    /// //
+    /// //#[derive(Debug, Serialize, Deserialize, PartialEq, Eq)]
+    /// //struct S {
+    /// //    addr: BDAddr,
+    /// //}
```

Comment:
I would like to write further examples in this style. However, when I try to compile this I get an error _crate not found_. even with the `use serde_cr as serde;`. Has anyone an idea how to fix this? (Yes, I did use `--all-features` to enable the `serde` feature.

Reply:
Okay got it myself